src/components/ThirdPersonCamera.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+#include <BsPrerequisites.h>
+#include <Scene/BsComponent.h>
+#include <RTTI/RTTIUtil.hpp>
+
+namespace REGoth
+{
+  class Character;
+  using HCharacter = bs::GameObjectHandle<Character>;
+
+  class ThirdPersonCamera;
+  using HThirdPersonCamera = bs::GameObjectHandle<ThirdPersonCamera>;
+
+  /**
+   * Third-Person Camera following a character. This is the camera used in normal gameplay.
+   *
+   * Cameras in Gothic seem to be a rather complex story. There are a number of camera
+   * script-instances defined in CAMERA.DAT, for example:
+   *
+   *   INSTANCE CamModRun(CCamSys_Def)
+   *   {
+   *     bestRange 		 = 2.5 ;
+   *     minRange  		 = 1.4 ;
+   *     maxRange  		 = 6.0 ;
+   *     bestElevation = 25.0 ;
+   *     minElevation	 = 0.0;
+   *     maxElevation	 = 89.0;
+   *     bestAzimuth	 = 0.0;
+   *     minAzimuth		 = -90.0;
+   *     maxAzimuth		 = 90.0;
+   *     rotOffsetX		 = 20.0;
+   *     rotOffsetY		 = 0.0;
+   *   };
+   */
+  class ThirdPersonCamera : public bs::Component
+  {
+  public:
+    ThirdPersonCamera(const bs::HSceneObject& parent);
+    virtual ~ThirdPersonCamera();
+
+    /**
+     * Lets the camera follow the given character.
+     */
+    void follow(HCharacter character);
+
+    /**
+     * Applies the given delta to the cameras distance. This can be used to zoom in the
+     * camera using the mousewheel, for example.
+     *
+     * There is no direct setter for this distance, as the distance has to adapt to
+     * certain situations during gameplay.
+     *
+     * @param  delta  Positive values will move the camera further away from the target,
+     *                Negative values will move it closer.
+     */
+    void modifyDistance(float delta);
```

Comment:
Misses the implementation in the `*.cpp`. Maybe add an empty body there with a `TODO` as we will certainly need this.

Reply:
You probably also had in mind the case where the camera is moved towards the character when back-facing a wall or any other object that is closer than the camera distance. Maybe we could use bs::f's physics to solve this by giving the camera SO a small collider and constraining the motion of this collider to the axis between camera and player. Think of a spring.

Reply:
Right, seems like I just forgot to implement that. In the original engine, the camera can be blocked by walls too, but they do it via a raycast as far as I know.

Reply:
By the way, that method was more for zooming via the scrollwheel.